v2/of.go

```diff
@@ -57,6 +57,10 @@ func (o OfSlice[T]) FirstOr(defaultValue T) T {
 	return FirstOr(o.Result, defaultValue)
 }
 
+func (o OfSlice[T]) FlatMap(fn func(T) []T) OfSlice[T] {
```

Comment:
Missing `Flat`

Reply:
`Flat` works with a 2-dimensional array, I'm not sure it should be part of the `OfSlice` type

Reply:
Yes, it should be part of `OfSlice` since it deals with slices.

Reply:
It's impossible to add `Flat` to the `OfSlice` type because `Result` is a one-dimensional slice (`[]T`) but `Flat` works with a two-dimensional slice (`[][]T`) so this does not work:

```
func (o OfSlice[T]) Flat() []T {
	return OfSlice[T]{Flat(o.Result)}
}
```

if you mean another way to add it to the `OfSlice` type I'll be thankful for your help

Reply:
Ah, I see what you mean. That's fair - you can leave it out.